src/intrinsics/ecma262/Array.js

```diff
@@ -85,7 +86,12 @@ export default function (realm: Realm): NativeFunctionValue {
         // c. Let intLen be 1.
         intLen = 1;
       } else { // 7. Else,
-        len = len.throwIfNotConcreteNumber();
+        // len = recoverableError(() => len.throwIfNotConcreteNumber(), ()=>new NumberValue(realm, 0));
+        len = recoverableError(
+          realm,
+          ()=>len.throwIfNotConcreteNumber(),
+          ()=>new NumberValue(realm, 0));
+
```

Comment:
The comment was a remnant from when I was trying to model the helper function.
Regarding the helper function, my goal was to avoid requiring overly verbose code, since the way the code is currently structured relies on exceptions - which means that ```something is bad``` from above would translate into a try-catch
My original version had that try-catch explicitly in the code, and I introduced the helper function to make it easy to add support for more recoverable errors. I'm open to suggestions regarding other ways to handle this, or if you think writing the verbose version of the code wherever we want to enable handling recovery would be more fitting.